### PR TITLE
rework state management docs

### PR DESCRIPTION
**Pull request by HarshCasper**
## Introduction

This PR consolidates all our state management features such as:

- Cloud Pods
- Persistence 
- Export/Import State (CLI & Web App) 

Under one section in the User Guides: `State Management`. 

The primary aim is to help the users discover all aspects of taking snapshots and help them discover various features that would be helpful for them.

### CODE REVIEW

File: content/en/tutorials/reproducible-machine-learning-cloud-pods/index.md [hunk 1]
```diff
@@ -24,9 +24,9 @@ pro: true
...
 
-Cloud Pods is supported by both [LocalStack Pro](https://app.localstack.cloud/) and [LocalStack Community](https://github.com/localstack/localstack). Using [Community Cloud Pods]({{< ref "user-guide/cloud-pods/community" >}}), you get a limited experience saving and loading your LocalStack state in a Cloud Pod, only with the AWS services emulated in the Community Edition. With LocalStack Pro, you can utilize an extended CLI that allows you to inspect your Cloud Pods, version them using tags, and push them to the LocalStack platform for storage and collaboration.
+Cloud Pods is supported by both [LocalStack Pro](https://app.localstack.cloud/) and [LocalStack Community](https://github.com/localstack/localstack). Using [Community Cloud Pods]({{< ref "user-guide/state-management/community" >}}), you get a limited experience saving and loading your LocalStack state in a Cloud Pod, only with the AWS services emulated in the Community Edition. With LocalStack Pro, you can utilize an extended CLI that allows you to inspect your Cloud Pods, version them using tags, and push them to the LocalStack platform for storage and collaboration.
```
Comment (giograno) [inline content/en/tutorials/reproducible-machine-learning-cloud-pods/index.md:29]: This part is outdated now. There is no way to extract state without an hobby license. Probably we should also get rid of the Community Cloud Pods page.

**Review by giograno — CHANGES_REQUESTED** [review]
Thanks for taking the initiative of cleaning this section of the docs 👏 

As the comments below, I would add a brief explanation in the `State Management` parent page, listing the 3 ways we offer to achieve state management in LocalStack, namely snapshot persistence, state export/import, and cloud pods. 
We should explain well the differences, especially between the last 2, with pods being a more advanced version of state export/import since it offers a remote storage, a versioning system, and a merge approach (plus whatever feature we are going to build in the future), only offered from the >= Team license.

We could even move `cloud-pods-persistence.png` (with some rework, see below) in the parent page. Also adding the info of which plan could use what might be a good idea.

File: content/en/user-guide/state-management/_index.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+title: "State Management"
+weight: 10
+description: >
+  Cloud Pods provides a new way of collaborating in cloud application development workflows.
+aliases:
+  - /user-guide/tools/cloud-pods/
+  - /user-guide/cloud-pods/
+---
+
+State Management is awesome!
```
[the remark this answers is not in the document]
Comment (HarshCasper) [reply content/en/user-guide/state-management/_index.md:11]: Yes, was planning to do that after an initial review 😅

File: content/en/tutorials/reproducible-machine-learning-cloud-pods/index.md (hunk 1 above)
Comment (HarshCasper) [reply content/en/tutorials/reproducible-machine-learning-cloud-pods/index.md:29]: Yes, will fix!

**Comment by thrau** [comment]
hey @HarshCasper , what's going on with this PR? would be great if we could get it out with the 3.1 release

**Comment by HarshCasper** [comment]
Hi @giograno @thrau — I have taken the liberty of merging the docs along side the suggestions made earlier. I will further collect feedback and improve this further (especially the differences between different features!). Thank You 🙏

**Comment by trb** [comment]
I'm disappointed that you've removed any mention about persistence/pods/state management being paid features in the docs.

Persistence  was a deciding factor when I picked localstack a couple of years ago, and you've pulled the rug out from under me by removing any persistence options for the community edition.

But you're also omitting this fact from your documentation. If I were to assess tooling today, I might be misled into believing I'd get state management with your tool in the community edition. Until I'd try to actually use it, and then I get the aggravating message "👋 This feature is part of our paid offering!" (Please, do not use cute emojis or other attempts to make this seem less like a shake-down. This is going to waste hours and hours of my time, on top of figuring out why persistence isn't working anymore in the first place. I do not want to see any attempts at making this seem cute or lighthearted.)

Not mentioned the paywall outright just feels scummy to me. If you want to charge for persistence then whatever, if you want to take available features and put them behind the paywall, whatever. But at least own up to it and point it out in the documentation, instead of letting users find out once they've spent time setting everything up. Unless you're banking on people just paying up, with the sunk-cost fallacy kicking in. Which..yeah, feels scummy. The more I think about these changes the worse I feel.

**Comment by thrau** [comment]
thanks for your feedback Thomas @trb. i understand your frustration and we're working on improving both our docs and our communication around our paid offering.

please feel free to reach out to me directly on our community slack, i'd be happy to give you some information on free options of our pro edition.

**Comment by trb** [comment]
Hi Thomas,

thank you for getting back to me. I joined the Slack community and I'll reach out to you there.

**Pull request merged**